File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int ans = 0;
        int total = 0;
        int p1 = 1;
        
        for (int p2 = 1; p2 <= n; p2++) {
            total += p2;
            
            while (total > n) {
                total -= p1;
                p1++;
            }
            
            if (total == n) {
                ans++;
            }
        }
        return ans;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let arr = [];
    for (let i = 1; i <= n; i++) {
        arr.push(i);
    }
    
    let p1 = 0;
    let p2 = 0;
    let ans = 0;
    let sum = 0;
    
    while (p2 <= n) {
        sum += arr[p2++];
        
        while (sum > n) {
            sum -= arr[p1++];
        }
        
        if (sum == n) {
            ans++;
        }
    }
    return ans;
}